### ISSUE TITLE
On `linux` the `certificates` table appears to misreport  `not_valid_before` and `not_valid_after`

### ISSUE THREAD

**Issue by directionless** [osquery/osquery#8505]
We have a report from a customer that the `certificates` isn't reporting dates quite right. I have this snippet from the customer

```
# openssl x509 -text -noout -in /etc/ssl/certs/kolide.pem | grep -A1 Validity
        Validity
            Not Before: Dec 16 12:29:43 2024 GMT

osquery> SELECT unixepoch(),not_valid_before FROM certificates WHERE common_name = (SELECT CONCAT(uuid, ".[wonka.corp.mongodb.com](http://wonka.corp.mongodb.com/)") FROM osquery_info);
+-------------+------------------+
| unixepoch() | not_valid_before |
+-------------+------------------+
| 1734368254  | 1734370183       |
+-------------+------------------+
```

And I notice:
```
$ ruby -e 'puts Time.at 1734370183'
2024-12-16 12:29:43 -0500
```

So this seems like it's pretty clearly a timezone issue.

Digging through the code, this is coming from around https://github.com/osquery/osquery/blob/17d7e83aa42ff4163f164bec45ab05ba787895f9/osquery/tables/system/posix/openssl_utils.cpp#L89

@Smjert [points out](https://osquery.slack.com/archives/C08VA3XQU/p1734461143703889) that https://docs.openssl.org/3.0/man3/ASN1_TIME_set/ says:
> The ASN1_TIME_normalize() function converts an ASN1_GENERALIZEDTIME or ASN1_UTCTIME into a time value that can be used in a certificate. It should be used after the ASN1_TIME_set_string() functions and before ASN1_TIME_print() functions to get consistent (i.e. GMT) results.

We don't call it (in convertAsn1TimeToString), so it's possible that the extracted date is not GMT beforehand, while we have a comment that says that the print function will always return a GMT time
https://github.com/osquery/osquery/blob/6ac7ffb1941703cd1317822c89357fe823b353d2/osquery/tables/system/posix/openssl_utils.cpp#L80-L82
might want to check that

**Comment by raaz252** [comment]
<img width="1034" height="370" alt="Image" src="https://github.com/user-attachments/assets/47c9bc9b-bf0a-45d7-bdf8-408edeaf28f4" />
i tried reproducing this issue but it results in correct timezone.
am i missing something ?

**Comment by directionless** [comment]
I agree, this looks like it's been fixed somewhere in the last couple years. Thank you!
